agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/configuration/SnippetConfiguration.java

```diff
@@ -20,19 +19,27 @@ public class SnippetConfiguration {
 
   // visible for testing
   static String readSnippet() {
-    try {
-      Path path = getSnippetFilePath("javascript-snippet.txt");
-      byte[] bytes = Files.readAllBytes(path);
-      return new String(bytes, StandardCharsets.UTF_8);
-    } catch (IOException | URISyntaxException e) {
-      LOGGER.error("Failed to read javascript-snippet file", e);
-    }
-    return "";
-  }
-
-  private static Path getSnippetFilePath(String resourceName) throws URISyntaxException {
     ClassLoader classLoader = SnippetConfiguration.class.getClassLoader();
-    return Paths.get(classLoader.getResource(resourceName).toURI());
+    String resourceName = "javascript-snippet.txt";
+    InputStream inputStream = classLoader.getResourceAsStream(resourceName);
```

Comment:
early return will help reduce the amount of indentation below

```suggestion
    InputStream inputStream = classLoader.getResourceAsStream(resourceName);
    if (inputStream == null) {
      LOGGER.error("Resource not found: " + resourceName);
      return "";
    }
```